sql/executor.go

```diff
@@ -192,59 +190,77 @@ func (e *Executor) execStmt(stmt parser.Statement, params parameters, planMaker
 			return err
 		}
 
-		result.Columns = plan.Columns()
-		for plan.Next() {
-			values := plan.Values()
-			row := driver.Result_Row{Values: make([]driver.Datum, 0, len(values))}
-			for _, val := range values {
-				if val == parser.DNull {
-					row.Values = append(row.Values, driver.Datum{})
-				} else {
-					switch vt := val.(type) {
-					case parser.DBool:
-						row.Values = append(row.Values, driver.Datum{
-							Payload: &driver.Datum_BoolVal{BoolVal: bool(vt)},
-						})
-					case parser.DInt:
-						row.Values = append(row.Values, driver.Datum{
-							Payload: &driver.Datum_IntVal{IntVal: int64(vt)},
-						})
-					case parser.DFloat:
-						row.Values = append(row.Values, driver.Datum{
-							Payload: &driver.Datum_FloatVal{FloatVal: float64(vt)},
-						})
-					case parser.DBytes:
-						row.Values = append(row.Values, driver.Datum{
-							Payload: &driver.Datum_BytesVal{BytesVal: []byte(vt)},
-						})
-					case parser.DString:
-						row.Values = append(row.Values, driver.Datum{
-							Payload: &driver.Datum_StringVal{StringVal: string(vt)},
-						})
-					case parser.DDate:
-						wireTimestamp := driver.Timestamp(vt.Time)
-						row.Values = append(row.Values, driver.Datum{
-							Payload: &driver.Datum_DateVal{
-								DateVal: &wireTimestamp,
-							},
-						})
-					case parser.DTimestamp:
-						wireTimestamp := driver.Timestamp(vt.Time)
-						row.Values = append(row.Values, driver.Datum{
-							Payload: &driver.Datum_TimeVal{
-								TimeVal: &wireTimestamp,
-							},
-						})
-					case parser.DInterval:
-						row.Values = append(row.Values, driver.Datum{
-							Payload: &driver.Datum_IntervalVal{IntervalVal: vt.Nanoseconds()},
-						})
-					default:
-						return fmt.Errorf("unsupported result type: %s", val.Type())
+		switch stmt.StatementType() {
+		case parser.DDL:
+			result.Union = &driver.Response_Result_DDL_{DDL: &driver.Response_Result_DDL{}}
+		case parser.RowsAffected:
+			resultRowsAffected := driver.Response_Result_RowsAffected{}
+			result.Union = &resultRowsAffected
+			for plan.Next() {
+				resultRowsAffected.RowsAffected++
+			}
+
+		case parser.Rows:
+			resultRows := &driver.Response_Result_Rows{
+				Columns: plan.Columns(),
+			}
+
+			result.Union = &driver.Response_Result_Rows_{
+				Rows: resultRows,
+			}
+			for plan.Next() {
+				values := plan.Values()
+				row := driver.Response_Result_Rows_Row{Values: make([]driver.Datum, 0, len(values))}
+				for _, val := range values {
+					if val == parser.DNull {
+						row.Values = append(row.Values, driver.Datum{})
+					} else {
```

Comment:
could you save a level of indentation here using `continue` in the `DNull` case?